sdk/search/azure-search/src/main/java/com/azure/search/SearchIndexAsyncClient.java

```diff
@@ -0,0 +1,736 @@
+     *
+     * @param documents collection of documents to be merged, if exists, otherwise uploaded
+     * @return response containing the document index result.
+     */
+    public Mono<Response<IndexDocumentsResult>> mergeOrUploadDocumentsWithResponse(Iterable<?> documents) {
+        return withContext(context -> mergeOrUploadDocumentsWithResponse(documents, context));
+    }
+
+    @SuppressWarnings("unchecked")
+    Mono<Response<IndexDocumentsResult>> mergeOrUploadDocumentsWithResponse(Iterable<?> documents, Context context) {
+        IndexBatch<?> batch = buildIndexBatch(documents, IndexActionType.MERGE_OR_UPLOAD);
+        return this.indexWithResponse(batch, context);
+    }
+
+    /**
+     * Deletes a collection of documents from the target index
+     *
+     * @param documents collection of documents to delete from the target Index.
+     * @return document index result.
+     */
+    public Mono<IndexDocumentsResult> deleteDocuments(Iterable<?> documents) {
+        return this.deleteDocumentsWithResponse(documents)
+            .map(Response::getValue);
+    }
+
+    /**
+     * Deletes a collection of documents from the target index
+     *
+     * @param documents collection of documents to delete from the target Index.
+     * @return response containing the document index result.
+     */
+    public Mono<Response<IndexDocumentsResult>> deleteDocumentsWithResponse(Iterable<?> documents) {
+        return withContext(context -> deleteDocumentsWithResponse(documents, context));
+    }
+
+    @SuppressWarnings("unchecked")
+    Mono<Response<IndexDocumentsResult>> deleteDocumentsWithResponse(Iterable<?> documents, Context context) {
+        IndexBatch<?> batch = buildIndexBatch(documents, IndexActionType.DELETE);
+        return this.indexWithResponse(batch, context);
+    }
+
+    /**
+     * Gets Client Api Version.
+     *
+     * @return the apiVersion value.
+     */
+    public SearchServiceVersion getApiVersion() {
+        return this.apiVersion;
+    }
+
+    /**
+     * Gets the endpoint for the Azure Cognitive Search service.
+     *
+     * @return the endpoint value.
+     */
+    public String getEndpoint() {
+        return this.endpoint;
+    }
+
+    /**
+     * Gets the number of documents
+     *
+     * @return the number of documents.
+     */
+    public Mono<Long> getDocumentCount() {
+        return this.getDocumentCountWithResponse()
+            .map(Response::getValue);
+    }
+
+    /**
+     * Gets the number of documents
+     *
+     * @return response containing the number of documents.
+     */
+    public Mono<Response<Long>> getDocumentCountWithResponse() {
+        return withContext(this::getDocumentCountWithResponse);
+    }
+
+    Mono<Response<Long>> getDocumentCountWithResponse(Context context) {
+        return restClient
+            .documents()
+            .countWithRestResponseAsync(context)
+            .map(Function.identity());
```

Comment:
Is there a need for this `map` given that it will just return the input value?

Reply:
yes since countWithRestResponseAsync returns a Mono of SimpleResponse (autogenerated code). The map(Function.identity()) allows returning a Mono of Response